index.js

```diff
@@ -31,10 +44,10 @@ exports.init = function (ssb, config) {
           const value = graph[source][dest]
           // Only if I am the `source` and `value >= 0`, request replication
```

Comment:
This comment seems outdated, should make make it clear that everything related to self is handled by req manager including blocks.

Reply:
I'm trying to understand the code here. So `hopStream` is from your point of view and graphstream is the graph in general so could be between 2 nodes A and B unrelated to your graph. Is that corrrect? And if so I'm wondering why we are handling source == self here, when it seems already handled in hopStream?

Reply:
> is the graph in general so could be between 2 nodes A and B unrelated to your graph. Is that corrrect?

Yes

> If so, I'm wondering why we are handling source == self here, when it seems already handled in hopStream?

I think the idempotency helps us here, in that it's not a problem to deal an edge twice. And I think why it's like this is that in the past when I was refactoring bermuda, for some reason (that I can't remember), I **had to** to have these seemingly overlapping APIs, otherwise it wouldn't work. But I can try removing this self-replicate from graphStream and leave it only for hopsStream, it might work just fine. Gladly we have tests.

Reply:
Sounds like a good plan :)